katz/src/main/kotlin/katz/instances/NonEmptyVectorMonad.kt

```diff
@@ -0,0 +1,32 @@
+package katz
+
+import java.util.Vector
+
+interface NonEmptyVectorMonad : Monad<NonEmptyVector.F> {
+    override fun <A> pure(a: A): HK<NonEmptyVector.F, A> = NonEmptyVector.of(a)
+
+    override fun <A, B> flatMap(fa: HK<NonEmptyVector.F, A>, f: (A) -> HK<NonEmptyVector.F, B>): HK<NonEmptyVector.F, B> =
+            fa.ev().flatMap { f(it).ev() }
+
+    @Suppress("UNCHECKED_CAST")
+    private tailrec fun <A, B> go(buf: Vector<B>, f: (A) -> HK<NonEmptyVector.F, Either<A, B>>, v: NonEmptyVector<Either<A, B>>): Unit =
```

Comment:
we had trouble with tailrec in the past with nested when expressions in a tailrec funciton.
One solution, if I not remember wrong, was move the tailrec function outside the class into an extension one. in [Free](https://github.com/FineCinnamon/Katz/blob/master/katz/src/main/kotlin/katz/free/Free.kt#L31) we have it in that way with a test to ensure that.

please could you add a test to ensure NonEmptyVectorMonad it's stack safe?